### PR TITLE
shim-unsigned: update to shim-15.3 release

### PR DESCRIPTION
**Pull request by christopherco**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

In preparation for supporting UEFI Secure Boot, update shim to shim-15.3 release. This release incorporates all the latest fixes and implements the generation-based revocation scheme, also known as Secure Boot Advanced Targeting (SBAT).

###### Change Log  

- Update to shim-15.3 release. Remove all previous patches. They are incorporated in latest shim-15.3 release
- Update embedded cert
- Add Mariner SBAT version data

###### Does this affect the toolchain?  

NO

###### Test Methodology

Local build

### CODE REVIEW

File: SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec [hunk 1]
```diff
@@ -1,83 +1,19 @@
...
+Source1:        sbat.csv.in
+Source100:      cbl-mariner-ca-20210127.der
+BuildRequires:  dos2unix
+BuildRequires:  vim-extra
 ExclusiveArch:  aarch64
 
-BuildRequires:  gnu-efi
-BuildRequires:  gnu-efi-devel
-
-Patch0001: 0001-Make-some-things-dprint-instead-of-console_print.patch
```
Comment (PawelWMS) [inline SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec:17]: I love it when so many files are being removed.:)

File: SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec [hunk 2]
```diff
@@ -87,21 +23,31 @@ loading.
 
 %prep
 %autosetup -n shim-%{version} -p1
+# shim Makefile expects vendor SBATs to be in data/sbat.<vendor>.csv
+sed -e "s,@@VERSION@@,%{version}-%{release},g" %{SOURCE1} > ./data/sbat.mariner.csv
```
Comment (PawelWMS) [inline SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec:27]: Isn't Microsoft the vendor in this case? This way we could also re-use the `%{vendor}` macro.;)

File: SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec [hunk 3]
```diff
@@ -1,83 +1,19 @@
...
-Source1:        cbl-mariner-ca.der
+Source0:        https://github.com/rhboot/shim/releases/download/%{version}/shim-%{version}.tar.bz2
+Source1:        sbat.csv.in
+Source100:      cbl-mariner-ca-20210127.der
+BuildRequires:  dos2unix
+BuildRequires:  vim-extra
 ExclusiveArch:  aarch64
 
-BuildRequires:  gnu-efi
-BuildRequires:  gnu-efi-devel
```
Comment (PawelWMS) [inline SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec:15]: Double-checking: are these two packages no longer needed?

Comment (christopherco) [reply SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec:15]: Correct, those packages are no longer needed. Long story but shim-15.3 now git submodules a forked version of gnu-efi source code which contain some fixes. See https://github.com/rhboot/shim/commit/c61bfdc8a34fea1c6ea97ac0fb9c99f34f980381

File: SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec (hunk 2 above)
Comment (christopherco) [reply SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec]: Fixed

Comment (christopherco) [reply SPECS/shim-unsigned-aarch64/shim-unsigned-aarch64.spec]: Actually, I will revert to using a hardcoded name. Linter is complaining that Vendor isn't `Vendor: Microsoft Corporation` so I cannot use %{vendor} in the file name...

**Review by PawelWMS — APPROVED** [review]

**Comment by christopherco** [comment]
Moving to draft. A new shim-15.4 release is coming soon.

**Comment by christopherco** [comment]
shim-15.3 should not be used. Shim-15.4 update now posted at #819

**Pull request closed without merging**